src/redux/user/user.sagas.js

```diff
@@ -1,36 +1,172 @@
+  REGISTER_USER
+} from './user.types';
 import { setItems } from '../../utils/client';
+import { SHOW_AFTER } from '../../configs/index';
 
-export const loginUser = (payload) => {
+export const loginUser = (data) => {
   const query = ` 
-  mutation login($user: UserInput!) {
+  mutation mutation($user: LoginInput!){
   loginUser(
-    user: $user
+    loginInput: $user
   ) {
     purchasedProducts
     orders
     token
-    id
+    _id
+    email
+    firstName
   }
 }
   `;
-  return setItems(query, payload);
+  return setItems(query, data);
+};
+
+export const confirmUser = (token) => {
+  const query = ` 
+  mutation confirmUser($token: String!){
+    confirmUser(token: $token)
+  }
+  `;
+  return setItems(query, { token });
+};
+
+export const recoverUser = (data) => {
+  const query = ` 
+  mutation recovery($email: String!, $language: Int!){
+    recoverUser(email: $email, language: $language)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const resetPassword = (data) => {
+  const query = ` 
+  mutation reset($password: String!, $token: String!){
+    resetPassword(password: $password, token: $token)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const checkIfTokenIsValid = (data) => {
+  const query = ` 
+  mutation checkToken($token: String!){
+  checkIfTokenIsValid(token: $token)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const registerUser = (data) => {
+  const query = ` 
+  mutation register($user: userRegisterInput!, $language: Int!){
+  registerUser(
+    user: $user,
+    language: $language
+  ) {
+    email
+  }
+  }
+  `;
+  return setItems(query, data);
 };
 
 export function* handleUserLoad({ payload }) {
   try {
-    yield put(setUserLoading());
+    yield put(setUserLoading(true));
     const user = yield call(loginUser, payload);
     yield put(setUser(user.data.loginUser));
+    yield put(setUserLoading(true));
     yield put(push('/'));
   } catch (error) {
     yield put(setUserError(error.message.replace('GraphQL error: ', '')));
   }
 }
 
+export function* handleUserConfirm({ payload }) {
+  try {
+    yield put(resetState());
+    yield put(setUserLoading(true));
+    yield call(confirmUser, payload);
+    yield put(setUserLoading(false));
+  } catch (error) {
+    yield put(setUserError(error.message.replace('GraphQL error: ', '')));
+  }
+}
+
+export function* handleUserRecovery({ payload }) {
+  try {
+    yield put(resetState());
+    yield put(setUserLoading(true));
+    yield call(recoverUser, payload);
+    yield put(setUserRecovered(true));
+    yield delay(SHOW_AFTER);
+    yield put(push('/login'));
+  } catch (error) {
+    yield put(setUserError(error.message.replace('GraphQL error: ', '')));
+  }
+}
+
+export function* handlePasswordReset({ payload }) {
+  try {
+    yield put(resetState());
+    yield put(setUserLoading(true));
+    yield call(resetPassword, payload);
+    yield put(setPasswordReset(true));
+    yield delay(SHOW_AFTER);
```

Comment:
Please, rename `SHOW_AFTER` to something lie `REDIRECT_TIMEOUT`, it is much more suitable name for that stuff

Reply:
renamed